### PR TITLE
Add "when" JSON conversion test.

### PR DESCRIPTION
**Pull request by abayer**
We need to make sure we've got a JSON conversion test for `when`, and it's good to have a visible example of the syntax for others to see.

cc @reviewbybees esp @michaelneale @rsandell

### CODE REVIEW

**Comment by michaelneale** [comment]
:bee: when passes. Nice one.

File: pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java [hunk 1]
```diff
@@ -295,4 +295,14 @@ public void globalLibrarySuccess() throws Exception {
...
+
+    @Test
+    public void basicWhen() throws Exception {
+        prepRepoWithJenkinsfile("basicWhen");
+
+        WorkflowRun b = getAndStartBuild();
+        j.assertBuildStatusSuccess(j.waitForCompletion(b));
+        j.assertLogContains("[Pipeline] { (One)", b);
+        j.assertLogContains("[Pipeline] { (Two)", b);
+    }
```
Comment (rsandell) [inline pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java]: I think you also need to test that a stage is actually also skipped.

Comment (abayer) [inline pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java]: Fair enough. Will add one.

**Comment by rsandell** [comment]
:bee:

**Pull request merged**